doc/rados/operations/pg-repair.rst

```diff
@@ -1,4 +1,65 @@
+============================
 Repairing PG inconsistencies
 ============================
+Sometimes a placement group might become "inconsistent". To return the
+placement group to an active+clean state, you must first determine which 
+of the placement groups has become inconsistent and then run the "pg
+repair" command on it. This page contains commands for diagnosing placement
+groups and the command for repairing placement groups that have become
+inconsistent.
+
+.. highlight:: console
+
+Commands for Diagnosing Placement-group Problems
+================================================
+The commands in this section provide various ways of diagnosing broken placement groups.
+
+The following command provides a high-level (low detail) overview of the health of the ceph cluster::
+
+   # ceph health detail
+
+The following command provides more detail on the status of the placement groups::
+
+   # ceph pg dump --format=json-pretty
+
+The following command lists inconsistent placement groups::
+
+   # rados list-inconsistent-pg {pgid} 
+
+The following command lists inconsistent rados objects::
+
+   # rados list-inconsistent-obj {pgid} 
+ 
+The following command lists inconsistent snapsets in the given placement group::
+
+   # rados list-inconsistent-snapset {pgid}
+
+
+Commands for Repairing Placement Groups
+=======================================
+The form of the command to repair a broken placement group is::
+
+   # ceph pg repair {pgid}
+  
+Where "{pgid}" is the id of the affected placement group.
+
+For example::
+
+   # ceph pg repair 1.4
+
+More Information on Placement Group Repair
+==========================================
+Ceph stores and updates the checksums of objects stored in the cluster. When a scrub is performed on a placement group, the OSD attempts to choose an authoritative copy from among its replicas. Among all of the possible cases, only one case is consistent. After a deep scrub, Ceph calculates the checksum of an object read from the disk and compares it to the checksum previously recorded. If the current checksum and the previously recorded checksums do not match, that is an inconsistency. In the case of replica pools, any mismatch between the checksum of any replica of an object and the checksum of the authoritative copy means that there is an inconsistency.
+
+The "pg repair" command attempts to fix inconsistencies of various kinds. If "pg repair" finds an inconsisent placement group, it attempts to overwrite the digest of the inconsistent copy with the digest of the authoritative copy. If "pg repair" finds an inconsistent replica pool, it marks the inconsistent copy as missing. Recovery, in the case of replica pools, is beyond the scope of "pg repair".
+
+"pg repair" will not solve every problem. Ceph does not automatically repair placement groups when inconsistencies are found in them.
```

Comment:
@zdover23 Are you going to talk about non-replicated/bluestore scrub/repair elsewhere?  "For erasure coded and/or bluestore pools, Ceph will automatically repair if osd_scrub_auto_repair (config default false) is set to true and at most osd_scrub_auto_repair_num_errors (config default 5) errors are found."

Reply:
I have incorporated this change, nearly verbatim.